README.md

```diff
@@ -365,6 +365,32 @@ Of course, the situation in which a class grows this long is strongly discourage
   }
   ```
 
+- If the only goal is to match on the type of the object, do NOT expand fully all the arguments, as it makes refactoring more difficult and the code more error prone.
```

Comment:
If you are matching and using all the argument, it seems likely that adding a new argument should have you revisit that code to see if it needs to be handled.  I'm not sure if its a "pro" but its at least acceptable.